SECURITY.md

```diff
@@ -0,0 +1,14 @@
+# Security Policy
+
+## Supported Versions
+
+Security updates are applied only to the latest release.
```

Comment:
see #1808 - we don't really have release or version numbers  this might be better stated as 

```security updates are applied only to the repository itself.```

s/latest release/repository itself/

Reply:
this coincides with the [previous suggestion](https://github.com/publicsuffix/list/pull/1856#pullrequestreview-1629660124)

Reply:
got it. I've updated to the suggested sentence.

Reply:
Also removed the "supported versions" subtitle since it was not applicable anymore